src/main/java/seedu/address/logic/commands/DeleteShownCommand.java

```diff
@@ -0,0 +1,120 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import javafx.collections.ObservableList;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Deletes all persons in the shown list.
+ */
+public class DeleteShownCommand extends Command {
+    public static final String COMMAND_WORD = "deleteshown";
+
+    public static final String MESSAGE_SUCCESS = "Deleted %d shown person(s). Listing %d remaining person(s).";
+    public static final String MESSAGE_NO_PERSONS = "No persons to delete.";
+    public static final String MESSAGE_NO_FILTER =
+            "For safety, all persons cannot be deleted. Use the 'clear' command instead.";
+
+    /**
+     * Deletes all persons in the last shown list.
+     * @param model {@code Model} which the command should operate on.
+     * @return a command result in which the success message is displayed
+     * @throws CommandException if the list is empty
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
```

Comment:
Good job identifying that this is different from the already existing `delete` command